loadtest/control/simulcontroller/actions.go

```diff
@@ -870,3 +870,101 @@ func searchChannels(u user.User) control.UserActionResponse {
 		return control.UserActionResponse{Info: fmt.Sprintf("found %d channels", len(channels))}
 	})
 }
+
+func searchPosts(u user.User) control.UserActionResponse {
+	team, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	var words []string
+	var opts control.PostsSearchOpts
+	// This is an arbitrary limit on the number of words to search for.
+	// TODO: possibly use user analytics data to improve this.
+	count := 1 + rand.Intn(4)
+
+	// TODO: back the probability of these choices with real data.
+	if rand.Float64() < 0.2 {
+		user, err := u.Store().RandomUser()
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		opts.From = user.Username
+	}
+
+	if rand.Float64() < 0.2 {
+		channel, err := u.Store().RandomChannel(team.Id, store.SelectMemberOf|store.SelectNotDirect|store.SelectNotGroup)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		opts.In = channel.Name
+	}
+
+	timeOpts := []*time.Time{
+		&opts.On,
+		&opts.Before,
+		&opts.After,
+	}
+	if rand.Float64() < 0.2 {
+		to := timeOpts[rand.Intn(len(timeOpts))]
+		*to = time.Now().Add(-time.Duration(rand.Intn(7)) * time.Hour * 24)
```

Comment:
Why 7?

And also, this seems like a roundabout way to do things. Is there a reason we cannot just get a random number between 0 to 3 and update the right variable accordingly?

Reply:
> Why 7?

It's an arbitrary value like everything else in this kind of methods. I chose to go back not more than a week. hence the 7. This was to address @isacikgoz's review.

> And also, this seems like a roundabout way to do things. Is there a reason we cannot just get a random number between 0 to 3 and update the right variable accordingly?

It's the first thing I thought of. Maybe it was the [C](https://en.wikipedia.org/wiki/C_(programming_language)) coder inside me. I *simplified*.